WooCommerce/WooCommerceTests/ViewRelated/Orders/Order Creation/PaymentSection/Taxes/TaxEducationalDialogViewModelTests.swift

```diff
@@ -55,16 +53,15 @@ final class TaxEducationalDialogViewModelTests: XCTestCase {
 
     func test_wpAdminTaxSettingsURL_passes_right_url() {
         // Given
-        let sampleAdminURL = "https://testshop.com/wp-admin/"
-        let sessionManager = SessionManager.testingInstance
-        let site = Site.fake().copy(adminURL: sampleAdminURL)
-        sessionManager.defaultSite = site
-        let stores = MockStoresManager(sessionManager: sessionManager)
+        let wpAdminTaxSettingsURL = URL(string: "https://www.site.com/wp-admin/mock-taxes-settings")
+        let wpAdminTaxSettingsURLProvider = MockWPAdminTaxSettingsURLProvider(wpAdminTaxSettingsURL: wpAdminTaxSettingsURL)
 
-        viewModel = TaxEducationalDialogViewModel(orderTaxLines: [], taxBasedOnSetting: .customerShippingAddress, stores: stores)
+        viewModel = TaxEducationalDialogViewModel(orderTaxLines: [],
+                                                  taxBasedOnSetting: .customerShippingAddress,
+                                                  wpAdminTaxSettingsURLProvider: wpAdminTaxSettingsURLProvider)
 
-        let expectedURLString = sampleAdminURL + "admin.php?page=wc-settings&tab=tax"
-        XCTAssertEqual(viewModel.wpAdminTaxSettingsURL?.absoluteString, expectedURLString)
+        XCTAssertNotNil(wpAdminTaxSettingsURL)
```

Comment:
This one won't be nil since we're declaring it and is not Optional, perhaps we could test if the resultant optional URL property from the URLProvider is?
```suggestion
        XCTAssertNotNil(wpAdminTaxSettingsURLProvider.wpAdminTaxSettingsURL)
```
That said, perhaps is not necessary since we're testing it implicitly on the next line when asserting the URL.